packages/server/src/queries/complex/assets/market.ts

```diff
@@ -27,24 +26,17 @@ export type AssetMarketInfo = Partial<{
 const marketInfoCache = new LRUCache<string, CacheEntry>(DEFAULT_LRU_OPTIONS);
 /** Cached function that returns an asset with market info included. */
 export async function getMarketAsset<TAsset extends Asset>({
-  assetLists,
-  chainList,
   asset,
 }: {
-  assetLists: AssetList[];
-  chainList: Chain[];
   asset: TAsset;
 }): Promise<TAsset & AssetMarketInfo> {
   const assetMarket = await cachified({
     cache: marketInfoCache,
     key: `market-asset-${asset.coinMinimalDenom}`,
     ttl: 1000 * 60 * 5, // 5 minutes
     getFreshValue: async () => {
-      const [currentPrice, marketCap, assetMarketActivity, coingeckoCoin] =
-        await Promise.all([
-          getAssetPrice({ assetLists, chainList, asset }).catch((e) =>
-            captureErrorAndReturn(e, undefined)
-          ),
+      const [marketCap, assetMarketActivity, coingeckoCoin] = await Promise.all(
+        [
```

Comment:
Do we need to couple all 3 queries together and block them on each other? I think these could be split and fetched separately

Reply:
This allows them to be fetched concurrently. WDYM separately?

Reply:
Let's say one query times out for one or the other reason. Then, the entire operation/display is blocked.

By fetching data separately and displaying on-demand, we will maximize fault-tolerance and enable best-effort display

Reply:
To be clear, this is outside of the scope of this PR as it would require larger changes. However, I think these are easy performance and fault-tolerance wins that we can tackle

Reply:
The catch statements would still allow the promise to resolve, and the error would get logged in Sentry, and undefined would get returned. So, yes, `Promise.all` does throw if one of the promises throws, but since we use `catch()` on all the contained promises then we prevent them all from throwing.

Reply:
My comment is not limited to `Promise.all` constraints. What I'm trying to point out is that, for example, we don't need to block displaying asset market cap on market activity data.

As a result, these should be fetched via separate operations as opposed to one